pkg/printers/internalversion/printers.go

```diff
@@ -831,6 +838,10 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 				reason = "NotReady"
 			}
 		}
+		// change pod status back to "Error" if there is at least one container still reporting as "Error" status
+		if hasError && !hasErrorReason(reason, "Error", "Signal:", "ExitCode:") {
```

Comment:
A few questions about this part:
- This override would take precedence over the one above, and I suspect it probably shouldn't, at least in the first part of the condition wher the pod has the ready condition set to true.
- Do we really know for sure that the error message will have one of these prefixes, given that it can come straight from "container.State.Terminated.Reason"?
- Especially if we can't reliably identify the error message at this point, do we really want to be overriding, or should the loop above instead preserve (or even accumulate?) the error reason(s) found?

Reply:
You are right! I change this logic.